src/Support/SelectFields.php

```diff
@@ -98,12 +98,12 @@ public static function getSelectableFieldsAndRelations(
         }
 
         return function ($query) use ($with, $select, $customQuery, $requestedFields, $ctx): void {
+            $query->select($select);
+            $query->with($with);
+            
             if ($customQuery) {
                 $query = $customQuery($requestedFields['args'], $query, $ctx);
             }
-
-            $query->select($select);
-            $query->with($with);
```

Comment:
This is a BC break: previously, `$customQuery` could returned a very different/completely custom `$query` but now that's not possible anymore.

For the fullest flexibility, we would need both:
- a `$createCustomQuery` (i.e. the current functionality)
- a `$adaptQuery` (i..e. what you're proposing)

Reply:
Hmm,  I might have meant to use addSelect, let me modify that and see if that changes anything